components/schemes/Color/tests/unit/MDCTonalPaletteTests.m

```diff
@@ -40,15 +40,15 @@ - (void)testTonalPaletteEncodingDecoding {
 
   // Given
   NSArray<UIColor *> *tonalColors = @[
-    [UIColor colorWithRed:0.6f green:0.2f blue:0.3f alpha:1],
-    [UIColor colorWithRed:1 green:0.5f blue:0 alpha:1],
-    [UIColor colorWithRed:0.2f green:0.3f blue:0.3f alpha:1],
-    [UIColor colorWithRed:0.5f green:0.8f blue:0.6f alpha:1],
-    [UIColor colorWithRed:0.4f green:0.7f blue:0.4f alpha:1],
-    [UIColor colorWithRed:0.8f green:0.3f blue:0.9f alpha:1],
-    [UIColor colorWithRed:0.6f green:0.2f blue:0.3f alpha:1],
-    [UIColor colorWithRed:0 green:0.8f blue:1 alpha:1],
-    [UIColor colorWithRed:0.1f green:0.8f blue:0.3f alpha:1],
+    [UIColor colorWithRed:(float)0.6 green:(float)0.2 blue:(float)0.3 alpha:1],
```

Comment:
should these be CGFloat castings here and below?

Reply:
These need to explicitly be single-precision `float` values because it seems that UIColor either encodes or decodes its values as `float` (and not CGFloat). Rather than keep the `0.7f` format that we decided against, I switched these to `(float)` (along with some others elsewhere in the code).  I felt it was better to have an explicit cast to `float` from `double` rather than use the `0.7f` format so that we can easily detect regressions back to the old format in the future.  At the same time, explicit float values like these will stand-out in the code and be acknowledged as necessary exceptions.